pom.xml

```diff
@@ -599,131 +670,129 @@
-                        <configuration>
-                            <cloneProjectsTo>${project.build.directory}/it</cloneProjectsTo>
-                            <cloneClean>true</cloneClean>
-                            <settingsFile>src/it/settings.xml</settingsFile>
-                            <localRepositoryPath>${project.build.directory}/local-repo</localRepositoryPath>
-                            <postBuildHookScript>verify</postBuildHookScript>
-                            <addTestClassPath>true</addTestClassPath>
-                            <skipInvocation>${skipTests}</skipInvocation>
-                            <streamLogs>true</streamLogs>
-                            <invokerPropertiesFile>invoker.properties</invokerPropertiesFile>
-                            <environmentVariables>
-                                <MAVEN_OPTS>
-                                    -javaagent:${settings.localRepository}/org/jacoco/org.jacoco.agent/${jacoco.version}/org.jacoco.agent-${jacoco.version}-runtime.jar=destfile=${project.basedir}/target/jacoco.exec,append=true
-                                </MAVEN_OPTS>
-                            </environmentVariables>
-                        </configuration>
-                        <executions>
-                            <execution>
-                                <id>integration-tests</id>
-                                <goals>
-                                    <goal>install</goal>
-                                    <goal>run</goal>
-                                </goals>
-                            </execution>
+							<!-- Prepares a variable, jacoco.agent.ut.arg, that contains the info 
+								to be passed to the JVM hosting the code being tested. -->
+							<execution>
+								<id>prepare-ut-agent</id>
+								<goals>
+									<goal>prepare-agent</goal>
+								</goals>
+								<phase>process-test-classes</phase>
+							</execution>
+							<execution>
+								<id>default-report</id>
+								<goals>
+									<goal>report</goal>
+								</goals>
+								<phase>verify</phase>
+							</execution>
+							<execution>
+								<!-- Prepares a variable, jacoco.agent.it.arg, that contains the 
+									info to be passed to the JVM hosting the code being tested. -->
+								<id>prepare-it-agent</id>
+								<goals>
+									<goal>prepare-agent-integration</goal>
+								</goals>
+								<phase>pre-integration-test</phase>
+								<configuration>
+									<propertyName>jacoco.agent</propertyName>
+								</configuration>
+							</execution>
+							<execution>
+								<id>default-report-itest</id>
+								<goals>
+									<goal>report-integration</goal>
+								</goals>
+								<phase>verify</phase>
+							</execution>
                         </executions>
                     </plugin>
                 </plugins>
             </build>
+   			<dependencies>
+				<dependency>
+					<groupId>org.sonarsource.java</groupId>
```

Comment:
Any reason why we need this new dependency?

Reply:
you are right, that was used in older sonar-plugin version in order to setup surefire-plugin to use a listener. but is not necessary anymore.